data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/MutateMapper.java

```diff
@@ -64,54 +79,94 @@ public RenameCopyConfig(String fromKey, String toKey) {
             } else if(Objects.equals(name, "uppercase")) {
                 uppercases.addAll(((ArrayList<String>) attr.getAttributeValue().getValue()).stream()
                         .map(NestedSyntaxConverter::convertNestedSyntaxToJsonPointer).collect(Collectors.toList()));
+            } else if(Objects.equals(name, "lowercase")) {
+                lowercases.addAll(((ArrayList<String>) attr.getAttributeValue().getValue()).stream()
+                        .map(NestedSyntaxConverter::convertNestedSyntaxToJsonPointer).collect(Collectors.toList()));
+            } else if(Objects.equals(name, "trim")) {
+                trims.addAll(((ArrayList<String>) attr.getAttributeValue().getValue()).stream()
+                        .map(NestedSyntaxConverter::convertNestedSyntaxToJsonPointer).collect(Collectors.toList()));
+            } else if(Objects.equals(name, "gsub")) {
+                final List<String> array = ((ArrayList<String>) attr.getAttributeValue().getValue()).stream()
+                        .map(NestedSyntaxConverter::convertNestedSyntaxToJsonPointer).collect(Collectors.toList());
+                for(int i = 0; i < array.size(); i += 3) {
+                    final SubstituteConfig newConfig = new SubstituteConfig(array.get(i), array.get(i + 1), array.get(i + 2));
```

Comment:
Having to iterate and get all three pieces of the gsub out isn't fun. If someone unfamiliar with gsub saw this down the line, they would probably be confused. It could help them out if we break each part out into variables.

```
final String gsubSource = array.get(i);
final String gsubPatternToReplace = array.get(i + 1);
final String gsubStringToReplaceWith = array.get(i + 2);
```